torch/_inductor/codegen/cpp.py

```diff
@@ -1561,25 +1561,23 @@ def load(self, name: str, index: sympy.Expr):
 
         return self.cse.generate(self.loads, line)
 
-    def store(self, name, index, value, mode=None):
-        assert "buf" in name
-        assert mode is None
-        opt_ctx: OptimizationContext = get_current_node_opt_ctx()
-        var = self.args.output(name)
-        index = self.rename_indexing(index)
+    def get_vec_store_line(self, value, var, index, dtype):
+        """
+        Get a store line str that stores `value` into `var` at `index` of `dtype`.
+        :param value: Vectorized type templaterized on `dtype`.
```

Comment:
Do you want to static assert that `value` is indeed of type `Vectorized<dtype>` or nah?